numpy/core/src/multiarray/_multiarray_tests.c.src

```diff
@@ -2045,6 +2045,21 @@ run_casting_converter(PyObject* NPY_UNUSED(self), PyObject *args)
     return PyInt_FromLong(casting);
 }
 
+static PyObject *
+run_intp_converter(PyObject* NPY_UNUSED(self), PyObject *args)
+{
+    PyArray_Dims dims = {NULL, -1};
+    if (!PyArg_ParseTuple(args, "O&", PyArray_IntpConverter, &dims)) {
+        return NULL;
+    }
+    if (dims.len == -1) {
+        Py_RETURN_NONE;
+    }
+
+    PyObject *tup = PyArray_IntTupleFromIntp(dims.len, dims.ptr);
+    PyDimMem_FREE(dims.ptr);
```

Comment:
Just in case for your (or anyone else's) info, we also have a `npy_free_cache_dims_obj()` which puts the dims back into a cache. But I like this here, since its public API and documented like this anway.

Reply:
I tried that, but this is compiled as a separate module so isn't available